src/containers/FileEditorContainer.js

```diff
@@ -49,28 +52,28 @@ const FileEditorContainer = ({kfId, history, match}) => {
               },
             ]}
           >
-            {(updateFile, {_}) => {
-              return (
-                <FileEditor
-                  kfId={data.fileByKfId.kfId}
-                  name={fileNameInput}
-                  description={data.fileByKfId.description}
-                  fileType={fileType}
-                  selectFileType={selectFileType}
-                  onSubmit={e => onSubmit(e, updateFile)}
-                  onNameChange={e => setFileName(e.target.value)}
-                  onDescriptionChange={e => setFileDescription(e.target.value)}
-                />
-              );
-            }}
+            {(updateFile, {_}) =>
+              children
+                ? children({
```

Comment:
This is a bit confusing, typically `children` is any jsx element, but it looks like you are expecting it to be of a specific component and calling it functionally instead of using standard jsx syntax.
What was wrong with simply returning a `<FileEditor />` component?

Reply:
we needed a way to add content before or after the editor in the context of the modal. Removing it here also helped to not duplicate the submission logic. 
When the FileEditor is inside of the modal the submission/save functionality is handled by the Modals footer buttons and not the form itself, so by removing the FileEditor here you can wrap the Modal and pass that logic down without needed the modal to handle the state for it.

Also open to another approach to this.